File: 0055-jump-game/0055-jump-game.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let n = nums.length;
    if(n == 1) return true;
    let maxJump = nums[0];
    let i = 1;
    while(maxJump){
        if(i == n-1){
            return true;
        }
        maxJump--;
        maxJump = Math.max(maxJump,nums[i]);
        i++;
    }
    return false;
};

File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int maxJump = nums[0];
        int n = nums.size();
        if(n == 1) return true;
        int i = 1;
        while(maxJump){
            if(i == n-1) return true;
            maxJump--;
            maxJump = max(maxJump,nums[i]);
            i++;
        }
        return false;
    }
};